libs/core/tests/unit_tests/runnables/test_tracing_interops.py

```diff
@@ -168,3 +169,31 @@ async def my_parent_function(a: int) -> int:
             )
         last_dotted_order = dotted_order
         parent_run_id = id_
+
+
+@patch("langchain_core.tracers.langchain.get_client")
+@pytest.mark.parametrize("enabled", [None, True, False])
+@pytest.mark.parametrize("env", ["", "true"])
+def test_tracing_enable_disable(mock_get_client: MagicMock, enabled: bool, env: str):
+    mock_session = MagicMock()
+    mock_client_ = Client(session=mock_session, api_key="test")
+    mock_get_client.return_value = mock_client_
+
+    def my_func(a: int) -> int:
+        return a + 1
+
+    env_on = env == "true"
+    with patch.dict("os.environ", {"LANGSMITH_TRACING": env}):
+        with tracing_context(enabled=enabled):
+            RunnableLambda(my_func).invoke(1)
+
+        time.sleep(0.3)
```

Comment:
rather than use time.sleep in unit tests, should we just turn temporarily turn off auto-batching / use `wait_for_all_tracers`?

Reply:
ya